components/webui/imports/ui/SearchView/SearchResultsTimeline.jsx

```diff
@@ -0,0 +1,177 @@
+ * @param {TimelineBucket[]} timelineBuckets
+ * @param {function} onTimelineZoom
+ * @return {JSX.Element}
+ */
+const SearchResultsTimeline = ({
+    timelineConfig,
+    timelineBuckets,
+    onTimelineZoom,
+}) => {
+    if (null === timelineBuckets) {
+        return <></>;
+    }
+
+    const data = {
+        datasets: [
+            {
+                backgroundColor: "#4096a0",
+                barPercentage: 1.2,
+                borderColor: "#007380",
+                borderWidth: 2,
+                minBarLength: 5,
+
+                data: adaptTimelineBucketsForChartJs(timelineBuckets),
+            },
+        ],
+    };
+
+    const options = {
+        animation: {
+            duration: 100,
+        },
+        maintainAspectRatio: false,
+        responsive: true,
+
+        scales: {
+            x: {
+                type: "time",
+
+                min: timelineConfig.range.begin.valueOf(),
+                max: timelineConfig.range.end.valueOf(),
+                offset: false,
+
+                grid: {
+                    drawOnChartArea: false,
+                    color: "black",
+                    drawTicks: true,
+                    offset: true,
+                },
+                ticks: {
+                    maxRotation: 0,
+                    autoSkipPadding: 30,
+                    major: {enabled: true},
+                },
+                time: {
+                    displayFormats: {
+                        millisecond: "HH:mm:ss.SSS",
+                        second: "HH:mm:ss",
+                        minute: "HH:mm",
+                        hour: "HH:mm",
+                    },
+                    parser: (date) => {
+                        return dayjs.utc(date)
+                            .tz(dayjs.tz.guess(), true)
+                            .format();
+                    },
+                },
+            },
+            y: {
+                ticks: {
+                    autoSkip: true,
+                    autoSkipPadding: 10,
+                },
+            },
+        },
+
+        plugins: {
+            tooltip: {
+                callbacks: {
+                    title: (tooltipItems) => {
+                        const [{raw: {x}}] = tooltipItems;
+                        const bucketStartTime = dayjs.utc(x);
+                        const bucketEndTime = bucketStartTime
+                            .add(timelineConfig.bucketDuration);
+
+                        return `${bucketStartTime.format(DATETIME_FORMAT_TEMPLATE)} to\n` +
+                            `${bucketEndTime.format(DATETIME_FORMAT_TEMPLATE)}`;
+                    },
+                },
+
+                caretSize: 0,
+                intersect: false,
+                mode: "x",
+                xAlign: "left",
+                yAlign: "bottom",
+            },
+            zoom: {
+                zoom: {
+                    drag: {
+                        enabled: true,
```

Comment:
Shall we disable the chart while the query is in progress similar to how we disable the query input?

Reply:
I agree. At least the zoom should be disabled when the query is ongoing.